### PR TITLE
[Fixes #40] SplitPackages needs to fail with a summary of all failures

### LINKED ISSUE #40: SplitPackages needs to fail with a summary of all failures
Right now it looks like it fails at the first package that is missing or new. It makes it very difficult to work with when a bunch of csv changes have to be made.

### PR DESCRIPTION
**Pull request by javiercn**

### CODE REVIEW

**Comment by javiercn** [comment]
@pranavkm

**Comment by pranavkm** [comment]
:shipit: Check with @Eilon if it's ok to check it in now.

File: src/SplitPackages/Program.cs [hunk 1]
```diff
@@ -150,6 +157,15 @@ private int Execute()
             }
         }
 
+        private void WriteErrorMessage(List<string> errors)
+        {
+            var header = Enumerable.Repeat(
```
Comment (Eilon) [inline src/SplitPackages/Program.cs]: That's a pretty fancy way of saying `var x = new[] { "foo" };`

File: src/SplitPackages/Program.cs [hunk 2]
```diff
@@ -150,6 +157,15 @@ private int Execute()
...
+                "The list of packages in the source folder is inconsistent with the list of packages on the CSV file:",
```
Comment (Eilon) [inline src/SplitPackages/Program.cs]: `*in* the CSV file`

**Comment by Eilon** [comment]
Fine for RTM. Don't need this in RC2.

**Comment by Eilon** [comment]
Oh, though I guess, there's no difference for tools: the tools are shared amongst all branches right? But whatever, that's fine.

**Comment by javiercn** [comment]
@Eilon Take a look now.

File: src/SplitPackages/Program.cs [hunk 3]
```diff
@@ -150,6 +157,24 @@ private int Execute()
...
+                
+            var message = string.Join($"{Environment.NewLine}    ", header.Concat(errors));
+            
+            if(_ignoreErrors.HasValue())
+            {
+                Logger.LogWarning(message);                
+            }
+            else
+            {
+                Logger.LogError(message);
```
Comment (pranavkm) [inline src/SplitPackages/Program.cs]: Do you know if `ILogger` actually writes to the error stream? Our CI is configured to pop errors up to the summary screen so it's nicer than having to dig through build logs.

File: src/SplitPackages/Program.cs [hunk 4]
```diff
@@ -465,14 +490,14 @@ private void CopyPackages(IList<PackageItem> expandedPackages)
...
         {
-            if (_warningsAsErrors.HasValue())
+            if (_ignoreErrors.HasValue())
             {
-                throw new InvalidOperationException(message);
+                Logger.LogWarning(message);
             }
-
-            Logger.LogWarning(message);
+
```
Comment (pranavkm) [inline src/SplitPackages/Program.cs]: Ctrl k - d

Comment (javiercn) [inline src/SplitPackages/Program.cs]: git rebase HEAD~1 --whitespace=fix

Comment (pranavkm) [inline src/SplitPackages/Program.cs]: Ahh, you fixed it :cake:

**Comment by pranavkm** [comment]
:shipit:

**Pull request merged**